src/imperative/imperative_utils.h

```diff
@@ -453,7 +453,7 @@ inline void PushOperator(const OpStatePtr& state,
         << "for stateful operator " << op->name;
     CHECK(exec_type == ExecType::kSync || exec_type == ExecType::kAsync);
     Engine::Get()->PushSync(
-      [state, fcompute, inputs, outputs, requested, is_train, exec_type, mutate_idx, req](
+      [&state, fcompute, &inputs, &outputs, &requested, is_train, exec_type, &mutate_idx, &req](
```

Comment:
I don't think you can do this.

Reply:
I was wondering why they were passed by value. Is Push() synchronous? This was my understanding.

Reply:
That this lambda is executed before this Push() call returns? Now that I think about it I probably should have verified that but I just always thought this was the case.

Reply:
Yeah, I guess it isn't synchronous like its comment description says. Removed.

Reply:
PushSync means the lambda is sync, but push itself is async